synIndex.R

```diff
@@ -25,44 +34,119 @@ source('params.R')
 # Get bucket params and file list
 #############
 ## Get a list of all Objects in the PRE_ETL S3 bucket 
-s3lsBucketObjects(source_bucket = paste0('s3://', PRE_ETL_BUCKET,'/'),
-                  output_file = FILE_LIST_OUTPUT)
+# s3lsBucketObjects(source_bucket = paste0('s3://', PRE_ETL_BUCKET,'/'),
+#                   output_file = FILE_LIST_OUTPUT)
 
 # Get bucket params
 bucket_params <- list(uploadType='S3',
-                    concreteType='org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting',
-                    bucket=SOURCE_BUCKET)
+                      concreteType='org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting',
+                      bucket=PRE_ETL_BUCKET)
 
 # The above list is just to verify/reference, 
 # since we will replicate the structure locally
 # we will work with that
 
-## Get a list of all local files
-localFileList <- list.files(path = AWS_DOWNLOAD_LOCATION,
-                            all.files = TRUE, # get hidden files too
-                            recursive = TRUE, # get files inside sub-folders (if any)
-                            full.names = FALSE) # to get the directory path prepended to the file name
+# all folders inside the AWS_DOWNLOAD_LOCATION
+# When we create dataFileHandleId in synapse it will create some folders in the source S3 bucket
+# we don't need those
+localDirs <- list.dirs(path = AWS_DOWNLOAD_LOCATION, full.names = FALSE) 
+
+## The folders we want to show in synapse
+## This would depend on the how the data is organized in the main S3 INGRESS bucket
+FOLDERS_TO_SYNC_SYNAPSE <- c('adults',
+                             'pregnant',
+                             'pediatric') 
+
```

Comment:
What happens if the organization chagnes. Are you going to assert that the organization matches. IT is always good to fail early with clear error/logging messages in case of failure.

Reply:
If the organization changes inside (for eg., adults/v2 or pregnant/v3) it is replicated in synapse courtesy of synapse manifest. If we find other folders coming in say (for eg., adults-new-site) they won't be synced, unless explicitly specified.